src/components/views/PCI.js

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import {
+  AccordionSet,
+  Button,
+  Col,
+  ExpandAllButton,
+  Headline,
+  Layout,
+  Pane,
+  PaneMenu,
+  Row,
+  Spinner,
+} from '@folio/stripes/components';
+
+import { IfPermission } from '@folio/stripes/core';
+
+import {
+  Agreements,
+  AgreementsList,
+  PCICoverage,
+  PCIInfo,
+  ParentPackageDetails,
+  TitleInfo,
+} from '../EResourceSections';
+
+export default class PCI extends React.Component {
+  static propTypes = {
+    data: PropTypes.shape({
+      eresource: PropTypes.shape({
+        id: PropTypes.string,
+        name: PropTypes.string,
+        status: PropTypes.shape({
+          label: PropTypes.string,
+        }),
+      }).isRequired,
+      relatedEntitlements: PropTypes.array,
+    }),
+    handlers: PropTypes.shape({
+      onClose: PropTypes.func.isRequired,
+    }),
+    isLoading: PropTypes.bool,
+  }
+
+  state = {
+    sections: {
+      coverage: true,
+      pciagreements: true
+    },
+  }
+
+  getSectionProps = (id) => {
+    const { data } = this.props;
+
+    return {
+      data,
+      id,
+      pci: data.eresource,
+      onToggle: this.handleSectionToggle,
+      open: this.state.sections[id],
+    };
+  }
+
+  handleSectionToggle = ({ id }) => {
+    this.setState((prevState) => ({
+      sections: {
+        ...prevState.sections,
+        [id]: !prevState.sections[id],
+      }
+    }));
+  }
+
+  handleAllSectionsToggle = (sections) => {
+    this.setState({ sections });
+  }
+
+  renderLoadingPane = () => {
+    return (
+      <Pane
+        defaultWidth="45%"
+        dismissible
+        id="pane-view-pci"
+        onClose={this.props.handlers.onClose}
+        paneTitle={<FormattedMessage id="ui-agreements.loading" />}
+      >
+        <Layout className="marginTop1">
+          <Spinner />
+        </Layout>
+      </Pane>
+    );
+  }
+
+  renderEditMenu = () => (
+    <PaneMenu>
+      <IfPermission perm="ui-agreements.resources.edit">
+        <FormattedMessage id="ui-agreements.eresources.edit">
+          {ariaLabel => (
+            <Button
+              aria-label={ariaLabel}
+              buttonStyle="primary"
+              id="clickable-edit-pci"
+              marginBottom0
+            >
+              <FormattedMessage id="stripes-components.button.edit" />
+            </Button>
+          )}
+        </FormattedMessage>
+      </IfPermission>
+    </PaneMenu>
+  );
+
+  renderPCIInfo = (eresource) => <PCIInfo pci={eresource} />;
```

Comment:
You're really overusing these render methods. There's no need to break them out of the main `render` function for these examples since it just forces us to scroll back and forth a bunch and pass variables along. `renderFoo` methods are useful when you have some logic you need to do in a computational scope to determine what to render. That's not happening here.

Additionally, what ends up happening is you lose track of code, like you did above with the `renderEditMenu` method. Similarly, in PCIForm, you lost track of the `renderLoadingPane` method.